sakai-evaluation-impl/src/java/org/sakaiproject/evaluation/logic/externals/ExternalHierarchyLogicImpl.java

```diff
@@ -94,6 +119,209 @@ public void init() {
     }
 
     /*
+      * (non-Javadoc)
+     * @see org.sakaiproject.evaluation.logic.externals.ExternalHierarchyLogic#getSectionsUnderEvalGroup(java.lang.String)
+     */
+    public List<Section> getSectionsUnderEvalGroup( String evalGroupID )
+    {
+        List<Section> sections = new ArrayList<Section>();
+
+        if( evalGroupID == null )
+        {
+            throw new IllegalArgumentException( "evalGroupId cannot be null" );
+        }
+
+        if( !evalGroupID.startsWith( EvalConstants.GROUP_ID_SITE_PREFIX ) )
+        {
+            throw new IllegalArgumentException( "cannot determine sections of groupId='" + evalGroupID + "' (must be a site)" );
+        }
+
+        // Determine if the group ID is pointing to a single section
+        boolean isSingleSection = false;
+        if( evalGroupID.contains( EvalConstants.GROUP_ID_SECTION_PREFIX ) )
+        {
+            isSingleSection = true;
+        }
+
+        try
+        {
+            // Get the site ID
+            String siteID = evalGroupID.replace( EvalConstants.GROUP_ID_SITE_PREFIX, "" );
+
+            // If the evalGroup is pointing to a single section, get the single session, add it to the list and return
+            if( isSingleSection )
+            {
+                sections.add( courseManagementService.getSection( evalGroupID.substring( evalGroupID.indexOf( EvalConstants.GROUP_ID_SECTION_PREFIX ) + 9 ) ) );
```

Comment:
Constant again. If you are parsing these evalGroupIDs in multiple places it might be good to have utility methods to do the work.